src/core/datagram.c

```diff
@@ -31,6 +31,17 @@ QuicDatagramValidate(
     _In_ const QUIC_DATAGRAM* Datagram
     )
 {
+    QUIC_CONNECTION* Connection = QuicDatagramGetConnection(Datagram);
```

Comment:
Looks like you have a build error because `QuicDatagramGetConnection` doesn't take a const pointer. You need to fix that.